addons/mail/models/mail_thread.py

```diff
@@ -4507,7 +4508,31 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
             # sudo: mail.message.translation - discarding translations of message after editing it
             self.env["mail.message.translation"].sudo().search([("message_id", "=", message.id)]).unlink()
             payload["Message"]["translationValue"] = False
-        self.env["bus.bus"]._sendone(message._bus_notification_target(), "mail.record/insert", payload)
+        notifications = [[message._bus_notification_target(), "mail.record/insert", payload]]
+        if outdated_starred_partners:
+            # sudo: bus.bus: reading non-sensitive last id
+            bus_last_id = self.env["bus.bus"].sudo()._bus_last_id()
+            notifications.extend([
+                [
+                    partner,
+                    "mail.record/insert",
+                    {
+                        "Thread": {
+                            "id": "starred",
+                            "messages": [("DELETE"), {"id": empty_messages.ids}],
+                            "model": "mail.box",
+                            "counter": count,
+                            "counter_bus_id": bus_last_id,
+                        }
+                    },
+                ]
+                for partner, count in self.env["mail.message"]._read_group(
+                    [("starred_partner_ids", "in", outdated_starred_partners.ids)],
+                    groupby=["starred_partner_ids"],
+                    aggregates=["__count"],
+                )
```

Comment:
At 1st I thought `count` could take empty messages thus was wrong. But there's `_cleanup_side_records` that's invoked whenever a message becomes empty/deleted, which cleans `starred_partner_ids`. So this should be good.

Reply:
Yeah, its called explicitly above that's why I have to save starred partner beforehand :)